docs/artifact-repository-credentials.md

```diff
@@ -0,0 +1,46 @@
+# Artifact Repository Credentials
+
+![alpha](assets/alpha.svg)
+
+> v2.9 and after
+
+You can reduce duplication in your templates by configuring repository credentials that can be accessed by any workflow. This can also remove sensitive information from your templates.
```

Comment:
Check the ArtifactRepositoryRef feature: https://github.com/argoproj/argo/pull/1350
It might already accomplish this goal.

Reply:
Interesting, an alternative approach that could be more flexible.

Reply:
I think I'm going to take your cue on this. That approach covers two use cases - the Argo operator manages the creds, and the workflow user manages them in their own namespace.

Reply:
Now I've dug deeper for 30s, I can see it is a similar but different use case, a different repo for the default artifact storage for a workflow. I think we could have commonality.

Reply:
You can even have different configurations with same name, but in different namespaces. This way the same workflow can write to different repositories when submitted to different namespaces.